File: solutions/1302-delete-characters-to-make-fancy-string/solution.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder FancyString=new StringBuilder();
        FancyString.append(s.charAt(0));
        int frequency=0;
        for(int i=1;i<s.length();i++){
            if(s.charAt(i)==s.charAt(i-1)){
               frequency++;
            }
            else{
                frequency=0;
            }
            if(frequency==2){
                frequency--;
            }
            else{
                FancyString.append(s.charAt(i));
            }
        }

        return FancyString.toString();
    }
}

File: solutions/1302-delete-characters-to-make-fancy-string/solution.py
class Solution(object):
    def makeFancyString(self, s):
        freq=0
        res=""
        res+=s[0]
        for i in range(1,len(s)):
            if(s[i]==s[i-1]):
                freq+=1
            else:
                freq=0
            if(freq==2):
                freq-=1
            else:
                res+=s[i]
        return res
        
